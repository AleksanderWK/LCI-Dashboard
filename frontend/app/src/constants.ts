// Frequency in Hz (cycles per second)
export const FREQUENCY = 2;

// Maximum time interval to show on live charts in seconds
export const LIVE_CHART_RANGE = 30;

export enum Variable {
    CognitiveLoad = "cl",
    PerceivedDifficulty = "pd",
    Familiarity = "fam",
    InformationProcessingIndex = "ipi",
    PhysiologicalArousal = "pa",
    Engagement = "eng",
    PhysiologicalStress = "ps",
    EmotionalRegulation = "er",
    MotionStability = "ms",
    EnergySpentFatigue = "esf",
    EducationalSpecificEmotions = "ese"
}

export enum Device {
    EyeTracker = "Eye Tracker",
    Wristband = "Wristband",
    VideoBody = "Video of Body",
    VideoFace = "Video of Face"
}

export enum EyeTrackingDevice {
    Mobile = "Mobile",
    Stationary = "Stationary"
}

export interface MMDVariableInfo {
    name: string;
    device: Device;
    description: string;
    maxValue?: number;
    calculationTime?: number;
    enabled: boolean;
}

export const MMDVariables: {[key in Variable]: MMDVariableInfo} = {
    [Variable.CognitiveLoad]: {
        name: "Cognitive Load",
        device: Device.EyeTracker,
        description:
            "Cognitive load refers to the amount of information that the working memory of a person can hold at once. In an educational setting, it is the effort needed for a learner to learn new information.",
        calculationTime: 30,
        enabled: true
    },
    [Variable.PerceivedDifficulty]: {
        name: "Perceived Difficulty",
        device: Device.EyeTracker,
        description:
            "Perceived difficulty is the perception of the ease or difficulty of performing in an educational context.",
        maxValue: 100,
        enabled: true
    },
    [Variable.Familiarity]: {
        name: "Familiarity",
        device: Device.EyeTracker,
        description: "",
        enabled: false
    },
    [Variable.InformationProcessingIndex]: {
        name: "Information Processing Index",
        device: Device.EyeTracker,
        description:
            "The ratio of local (100) and global (0) information processing. Staring at one point is local processing, while rapidly scanning a larger area is global processing.",
        maxValue: 100,
        enabled: true
    },
    [Variable.PhysiologicalArousal]: {
        name: "Physiological Arousal",
        device: Device.Wristband,
        description:
            "Physiological Arousal is how strong the current emotions are. It is calculated by the average EDA in a time interval of 8 seconds.",
        calculationTime: 8,
        enabled: true
    },
    [Variable.Engagement]: {
        name: "Engagement",
        device: Device.Wristband,
        description:
            "Engagement is a measure of involvement and enthusiasm for the activity. It is calculated by the area under the graph of tonic EDA, in a time interval of 4 seconds.",
        calculationTime: 4,
        enabled: true
    },
    [Variable.PhysiologicalStress]: {
        name: "Physiological Stress",
        device: Device.Wristband,
        description:
            "Physioloigcal Stress is indicated by LF/HF, the relative strengths of the low frequency and high frequency waves of the heart pulse signal.",
        calculationTime: 300,
        enabled: true
    },
    [Variable.EmotionalRegulation]: {
        name: "Emotional Regulation",
        device: Device.Wristband,
        description: "",
        enabled: false
    },
    [Variable.MotionStability]: {
        name: "Motion Stability",
        device: Device.VideoBody,
        description: "",
        enabled: false
    },
    [Variable.EnergySpentFatigue]: {
        name: "Energy Spent/Fatigue",
        device: Device.VideoBody,
        description:
            "The amount of energy spent by the student, also an indicator of fatigue. This is determined by calculating the jerk, the third derivative of the position skeletal body.",
        calculationTime: 3,
        enabled: true
    },
    [Variable.EducationalSpecificEmotions]: {
        name: "Educational-specific Emotions",
        device: Device.VideoFace,
        description: "Emotions that the student is experiencing in an educational setting.",
        enabled: true
    }
};

export const emotionsColorMapper: {[key: string]: string} = {
    boredom: "#F15025",
    frustration: "#191919",
    confusion: "#FFE45E",
    delight: "#6782B7"
};

export const emotionsIndexMapper: {[key: string]: number} = {
    boredom: 0,
    frustration: 1,
    confusion: 2,
    delight: 3
};

export const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December"
];
