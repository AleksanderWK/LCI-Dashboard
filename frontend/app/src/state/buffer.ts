import {atom, atomFamily} from "recoil";
import {DataPoints} from "../pages/SessionView";

export const sessionCodesState = atom<string[]>({
    key: "sessionCodes",
    default: []
});

/*
 *  An atomFamily that stores buffer data for each session
 */
export const bufferState = atomFamily<{active: boolean; data: DataPoints[]}, string>({
    key: "buffer",
    default: {active: false, data: []}
});
