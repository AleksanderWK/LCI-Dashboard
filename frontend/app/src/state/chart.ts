import {atom} from "recoil";

/*
 *  An atom that stores the current callback function ID (increased by 1 for each new callback)
 */
export const callbackFunctionIdState = atom<number>({
    key: "callbackFunctionId",
    default: 0
});

interface CallbackFunctions {
    [key: number]: () => void;
}

/*
 *  An atom that stores all functions to be called on every chart update
 */
export const callbackFunctionsState = atom<CallbackFunctions>({
    key: "callbackFunctions",
    default: {}
});
