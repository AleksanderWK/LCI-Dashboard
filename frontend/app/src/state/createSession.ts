/*
 *  State for creating a session
 */

import {atom} from "recoil";

export interface CreateSessionValues {
    sessionName: string;
    studentId: string;
    eyeTracker: string;
    sessionCode: string;
    studentConnected: boolean;
}

export const createSessionValuesState = atom<CreateSessionValues>({
    key: "createSessionValues",
    default: {
        sessionName: "",
        studentId: "",
        eyeTracker: "",
        sessionCode: "",
        studentConnected: false
    }
});
