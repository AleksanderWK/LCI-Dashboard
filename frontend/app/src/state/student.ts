import {atom, selectorFamily} from "recoil";

export interface Student {
    _id: string;
    name: string;
}

/*
 *  A selectorFamily that returns the student object for a given student id
 */
export const studentState = selectorFamily<Student | undefined, string>({
    key: "student",
    get: (id) => ({get}) => {
        const students = get(studentsState);
        return students.find((student) => student._id === id);
    }
});

/*
 *  A selector that returns a list of all students
 */
export const studentsState = atom<Student[]>({
    key: "students",
    default: []
});
