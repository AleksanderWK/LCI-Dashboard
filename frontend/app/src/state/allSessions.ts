/*
 *  State used in the all sessions view
 */

import {atom, selector} from "recoil";
import {Variable} from "../constants";
import {EducationalSpecificEmotions, sessionsState, sessionVariableDataState} from "./session";

export interface allsessionsObject {
    sessionId: number;
    studentName: string;
    data: [number, number | EducationalSpecificEmotions][];
}

/*
 *  An atom that stores which variable is selected for the all sessions view
 */
export const selectedAllSessionVariableState = atom<Variable | null>({
    key: "selectedAllSessionVariable",
    default: null
});

/*
 *  A selector that returns array of objects that each contain student name and session data for
 *  the currently selected variable.
 */
export const allSessionsState = selector<allsessionsObject[]>({
    key: "allSessions",
    get: ({get}) => {
        const result: allsessionsObject[] = [];
        const sessions = get(sessionsState);
        const selectedVariable = get(selectedAllSessionVariableState);
        sessions.forEach((session) => {
            if (selectedVariable != null) {
                const data = get(sessionVariableDataState([selectedVariable, session._id]));
                result.push({sessionId: session._id, studentName: session.student.name, data: data});
            }
        });
        return result;
    }
});
