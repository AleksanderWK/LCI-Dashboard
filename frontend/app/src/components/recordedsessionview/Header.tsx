import {createStyles, makeStyles, Typography, IconButton, SvgIcon, SvgIconProps, Theme} from "@material-ui/core";
import logo from "../../assets/Images/LCI_logo.png";
import TimerIcon from "@material-ui/icons/Timer";

import PersonIcon from "@material-ui/icons/Person";
import EventIcon from "@material-ui/icons/Event";
import QueryBuilderIcon from "@material-ui/icons/QueryBuilder";
import {useRecoilState, useRecoilValue, useResetRecoilState} from "recoil";
import {
    currentRecordingInterval,
    RecordedSessionInfo,
    recordedSessionInfoState,
    selectedRecordedSessionIdState
} from "../../state/recordedSession";
import {useHistory} from "react-router";
import {useEffect} from "react";
import {ipcInvoke} from "../../ipc";
import {Session} from "../../state/session";
import {Student} from "../../state/student";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        container: {
            gridColumnStart: 1,
            gridColumnEnd: 4,
            gridRowStart: 1,
            gridRowEnd: 2,
            width: "100%",
            height: "100%",
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-between",
            alignItems: "center"
        },

        sessionInfoContainer: {
            marginLeft: "25px",
            height: "60px",
            display: "flex",
            flexDirection: "column",
            justifyContent: "space-between",
            width: "490px"
        },

        indicatorIcon: {
            width: "7.5px",
            height: "7.5px",
            borderRadius: "7.5px",
            backgroundColor: "#7BAF62"
        },

        btnGroup: {
            display: "grid",
            gridTemplateColumns: "50px 50px",
            gridTemplateRows: "50px",
            alignItems: "center",
            gap: theme.spacing(1),
            marginRight: "30px"
        },
        contentLeft: {
            marginLeft: "30px",
            display: "flex",
            alignItems: "center"
        },
        textColor: {
            color: theme.palette.text.default
        },
        infoRow: {
            display: "grid",
            gridTemplateColumns: "30px max-content 30px 30px max-content 30px 30px max-content 30px 30px max-content",
            gridTemplateRows: "30px",
            alignItems: "center",
            justifyItems: "center"
        },
        header: {
            width: "auto"
        }
    })
);

function AddChartIcon(props: SvgIconProps): JSX.Element {
    return (
        <SvgIcon {...props}>
            <path
                d="M29 4.5V7.5H24.5V12H21.5V7.5H17V4.5H21.5V0H24.5V4.5H29ZM24.5 25.5H3.5V4.5H12.5V1.5H3.5C1.85 1.5 0.5 2.85 0.5 4.5V25.5C0.5 27.15 1.85 28.5 3.5 28.5H24.5C26.15 28.5 27.5 27.15 27.5 25.5V16.5H24.5V25.5ZM18.5 16.5V22.5H21.5V16.5H18.5ZM12.5 22.5H15.5V10.5H12.5V22.5ZM9.5 22.5V13.5H6.5V22.5H9.5Z"
                fill="#535353"
            />
        </SvgIcon>
    );
}

function ExitIcon(props: SvgIconProps): JSX.Element {
    return (
        <SvgIcon {...props}>
            <path
                d="M15.135 23.385L17.25 25.5L24.75 18L17.25 10.5L15.135 12.615L19.005 16.5H4.5V19.5H19.005L15.135 23.385ZM28.5 4.5H7.5C5.835 4.5 4.5 5.85 4.5 7.5V13.5H7.5V7.5H28.5V28.5H7.5V22.5H4.5V28.5C4.5 30.15 5.835 31.5 7.5 31.5H28.5C30.15 31.5 31.5 30.15 31.5 28.5V7.5C31.5 5.85 30.15 4.5 28.5 4.5Z"
                fill="#535353"
            />
        </SvgIcon>
    );
}
export default function Header(): JSX.Element {
    const classes = useStyles();
    const history = useHistory();

    const recordedSessionId = useRecoilValue(selectedRecordedSessionIdState);
    const [recordedSessionInfo, setRecordedSessionInfo] = useRecoilState(recordedSessionInfoState);
    const resetInterval = useResetRecoilState(currentRecordingInterval);

    useEffect(() => {
        if (recordedSessionId) {
            ipcInvoke("getSession", recordedSessionId).then((session) => {
                const sessionInfo = session as Session;
                const studentID = sessionInfo.studentId;
                ipcInvoke("getUserByID", studentID).then((student) => {
                    const studentObj = student as Student;
                    const name = studentObj.name;
                    if (sessionInfo.endTime) {
                        const info: RecordedSessionInfo = {
                            sessionId: sessionInfo._id,
                            sessionName: sessionInfo.sessionName,
                            studentId: sessionInfo.studentId,
                            eyeTrackingDevice: sessionInfo.eyeTrackingDevice,
                            date: new Date(sessionInfo.startTime).toDateString().slice(4),
                            startTime: `${("0" + new Date(sessionInfo.startTime).getHours()).slice(-2)}:${(
                                "0" + new Date(sessionInfo.startTime).getMinutes()
                            ).slice(-2)}`,
                            duration: calcDuration(sessionInfo.startTime, sessionInfo.endTime),
                            studentName: name
                        };
                        setRecordedSessionInfo(info);
                    }
                });
            });
        }
    }, [recordedSessionId]);

    const calcDuration = (start: number, end: number) => {
        let distance = end - start;
        const hours = Math.floor(distance / 3600000);
        distance -= hours * 3600000;
        const minutes = Math.floor(distance / 60000);
        distance -= minutes * 60000;
        const seconds = Math.floor(distance / 1000);
        return `${hours}h ${minutes}m ${seconds}s`;
    };

    return (
        <div className={classes.container}>
            <div className={classes.contentLeft}>
                <img src={logo} alt="The LCI logo" />

                <div className={classes.sessionInfoContainer}>
                    <Typography variant={"h1"} noWrap className={classes.header}>
                        {recordedSessionInfo?.sessionName}
                    </Typography>

                    <div className={`${classes.infoRow}`}>
                        <PersonIcon className={classes.textColor} />
                        <Typography>{recordedSessionInfo?.studentName}</Typography>
                        <div></div>
                        <EventIcon className={classes.textColor} />
                        <Typography>{recordedSessionInfo?.date}</Typography>
                        <div></div>
                        <QueryBuilderIcon className={classes.textColor} />
                        <Typography>{recordedSessionInfo?.startTime}</Typography>
                        <div></div>
                        <TimerIcon className={classes.textColor} />
                        <Typography>{recordedSessionInfo?.duration}</Typography>
                    </div>
                </div>
            </div>
            <div className={classes.btnGroup}>
                <IconButton aria-label="add new chart">
                    <AddChartIcon viewBox="0 0 29 29" />
                </IconButton>
                <IconButton
                    aria-label="exit recording view"
                    onClick={() => {
                        resetInterval();
                        history.push("/");
                    }}
                >
                    <ExitIcon viewBox="3 3 29 29" />
                </IconButton>
            </div>
        </div>
    );
}
