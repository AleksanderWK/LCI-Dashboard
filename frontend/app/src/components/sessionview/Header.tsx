import React, {useEffect, useState} from "react";
import {createStyles, makeStyles, Typography, IconButton, Theme} from "@material-ui/core";
import TimerIcon from "@material-ui/icons/Timer";
import RecordingButton from "./RecordingButton";
import {selectedSessionState} from "../../state/session";
import {useRecoilValue, useSetRecoilState} from "recoil";
import {quitSessionPopupOpenState} from "../../state/popup";
import HeaderWrapper from "../common/HeaderWrapper";
import InfoItem from "../common/InfoItem";
import {AddChartIcon, CloseIcon} from "../common/Icons";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        indicatorContainer: {
            display: "grid",
            gridTemplateColumns: "8px max-content",
            gap: theme.spacing(1)
        },
        indicatorIcon: {
            alignSelf: "center",
            width: "8px",
            height: "8px",
            borderRadius: "50%",
            backgroundColor: "#7BAF62"
        }
    })
);

export default function Header(): JSX.Element {
    const classes = useStyles();

    const setQuitSessionPopupOpen = useSetRecoilState(quitSessionPopupOpenState);

    const selectedSessionInfo = useRecoilValue(selectedSessionState);
    const [duration, setDuration] = useState<string>("");

    useEffect(() => {
        // Set the current duration initially when the compnent loads
        setCurrentDuration();

        // Interval that updates the duration state every second
        let intervalId: NodeJS.Timeout | null = null;
        intervalId = setInterval(() => {
            setCurrentDuration();
        }, 1000);

        return function cleanup() {
            if (intervalId) clearInterval(intervalId);
        };
    }, [selectedSessionInfo, setCurrentDuration]);

    // This function finds the current duration based on startTime in selectedSessionInfo and by using the current time
    function setCurrentDuration() {
        if (selectedSessionInfo) {
            const d = new Date().getTime();
            let distance = d - selectedSessionInfo.startTime;
            const hours = Math.floor(distance / 3600000);
            distance -= hours * 3600000;
            const minutes = Math.floor(distance / 60000);
            distance -= minutes * 60000;
            const seconds = Math.floor(distance / 1000);
            setDuration(`${hours}:${("0" + minutes).slice(-2)}:${("0" + seconds).slice(-2)}`);
        }
    }

    return (
        <>
            {selectedSessionInfo && (
                <HeaderWrapper
                    title={selectedSessionInfo.sessionName}
                    infoBar={
                        <>
                            <div className={classes.indicatorContainer}>
                                <div className={classes.indicatorIcon}></div>
                                <Typography>{selectedSessionInfo.student.name}</Typography>
                            </div>
                            <InfoItem icon={<TimerIcon />} text={duration} />
                        </>
                    }
                    buttonGroup={
                        <>
                            <RecordingButton />
                            <IconButton aria-label="add new chart">
                                <AddChartIcon />
                            </IconButton>
                            <IconButton
                                aria-label="quit student session"
                                onClick={() => {
                                    setQuitSessionPopupOpen(true);
                                }}
                            >
                                <CloseIcon />
                            </IconButton>
                        </>
                    }
                />
            )}
        </>
    );
}
