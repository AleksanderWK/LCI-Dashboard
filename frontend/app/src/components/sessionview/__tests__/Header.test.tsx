import React from "react";
import ReactDOM from "react-dom";
import {RecoilRoot} from "recoil";
import Header from "../Header";
import {
    selectedSessionIdState,
    Session,
    sessionIdsState,
    sessionRecordingState,
    sessionState
} from "../../../state/session";
import {studentsState} from "../../../state/student";
import {EyeTrackingDevice} from "../../../constants";

const testSessionIds = [1, 2];

const testStudents = [
    {_id: "1", name: "John Doe"},
    {_id: "2", name: "Jane Smith"}
];

const testSessions: {[key: number]: Session} = {
    1: {
        _id: 1,
        sessionName: "Test1",
        studentId: "1",
        studentName: "John Doe",
        eyeTrackingDevice: EyeTrackingDevice.Mobile,
        startTime: 0,
        endTime: 100,
        sessionCode: "abc"
    },
    2: {
        _id: 2,
        sessionName: "Test2",
        studentId: "2",
        studentName: "Jane Smith",
        eyeTrackingDevice: EyeTrackingDevice.Stationary,
        startTime: 200,
        endTime: 300,
        sessionCode: "def"
    }
};

const testSelectedSessionId = 1;

const testRecording = {
    1: {status: false, startTime: null, recordingId: 0},
    2: {status: true, startTime: new Date(), recordingId: 0}
};

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <RecoilRoot
            initializeState={(snap) => {
                snap.set(sessionIdsState, testSessionIds);
                snap.set(studentsState, testStudents);
                snap.set(sessionState(1), testSessions[1]);
                snap.set(sessionState(2), testSessions[2]);
                snap.set(selectedSessionIdState, testSelectedSessionId);
                snap.set(sessionRecordingState(1), testRecording[1]);
                snap.set(sessionRecordingState(2), testRecording[2]);
            }}
        >
            <Header />
        </RecoilRoot>,
        div
    );
});

// Snapshot test not possible due to duration of mocked session will always be different (since current date is used)
/*
it("Header matches snapshot", () => {
    const {baseElement} = render(
        <RecoilRoot
            initializeState={(snap) => {
                snap.set(sessionIdsState, testSessionIds);
                snap.set(studentsState, testStudents);
                snap.set(sessionState(1), testSessions[1]);
                snap.set(sessionState(2), testSessions[2]);
                snap.set(selectedSessionIdState, testSelectedSessionId);
                snap.set(sessionRecordingState(1), testRecording[1]);
                snap.set(sessionRecordingState(2), testRecording[2]);
            }}
        >
            <Header />
        </RecoilRoot>
    );
    expect(baseElement).toMatchSnapshot();
});
*/
