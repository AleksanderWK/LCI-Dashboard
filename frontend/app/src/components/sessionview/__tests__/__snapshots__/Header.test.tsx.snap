// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStudentPopup matches snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-1"
    >
      <div
        class="makeStyles-contentLeft-5"
      >
        <img
          alt="The LCI logo"
          src="LCI_logo.png"
        />
        <div
          class="makeStyles-sessionInfoContainer-2"
        >
          <h1
            class="MuiTypography-root makeStyles-header-8 MuiTypography-h1 MuiTypography-noWrap"
          >
            Test1
          </h1>
          <div
            class="makeStyles-infoRow-7"
          >
            <div
              class="makeStyles-indicatorIcon-3"
            />
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              John Doe
            </p>
            <div />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-textColor-6"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 1H9v2h6V1zm-4 13h2V8h-2v6zm8.03-6.61l1.42-1.42c-.43-.51-.9-.99-1.41-1.41l-1.42 1.42C16.07 4.74 14.12 4 12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9 9-4.03 9-9c0-2.12-.74-4.07-1.97-5.61zM12 20c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1"
            />
          </div>
        </div>
      </div>
      <div
        class="makeStyles-btnGroup-4"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </span>
            Start recording
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="add new chart"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              fill="none"
              focusable="false"
              height="29"
              viewBox="0 0 29 29"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M29 4.5V7.5H24.5V12H21.5V7.5H17V4.5H21.5V0H24.5V4.5H29ZM24.5 25.5H3.5V4.5H12.5V1.5H3.5C1.85 1.5 0.5 2.85 0.5 4.5V25.5C0.5 27.15 1.85 28.5 3.5 28.5H24.5C26.15 28.5 27.5 27.15 27.5 25.5V16.5H24.5V25.5ZM18.5 16.5V22.5H21.5V16.5H18.5ZM12.5 22.5H15.5V10.5H12.5V22.5ZM9.5 22.5V13.5H6.5V22.5H9.5Z"
                fill="#535353"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="quit student session"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              fill="none"
              focusable="false"
              height="29"
              viewBox="0 0 29 29"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.9408 8.55L14.25 12.2408L10.5592 8.55L8.55 10.5592L12.2408 14.25L8.55 17.9408L10.5592 19.95L14.25 16.2592L17.9408 19.95L19.95 17.9408L16.2592 14.25L19.95 10.5592L17.9408 8.55ZM14.25 0C6.36975 0 0 6.36975 0 14.25C0 22.1302 6.36975 28.5 14.25 28.5C22.1302 28.5 28.5 22.1302 28.5 14.25C28.5 6.36975 22.1302 0 14.25 0ZM14.25 25.65C7.96575 25.65 2.85 20.5343 2.85 14.25C2.85 7.96575 7.96575 2.85 14.25 2.85C20.5343 2.85 25.65 7.96575 25.65 14.25C25.65 20.5343 20.5343 25.65 14.25 25.65Z"
                fill="#535353"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Header matches snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-1"
    >
      <div
        class="makeStyles-contentLeft-5"
      >
        <img
          alt="The LCI logo"
          src="LCI_logo.png"
        />
        <div
          class="makeStyles-sessionInfoContainer-2"
        >
          <h1
            class="MuiTypography-root makeStyles-header-8 MuiTypography-h1 MuiTypography-noWrap"
          >
            Test1
          </h1>
          <div
            class="makeStyles-infoRow-7"
          >
            <div
              class="makeStyles-indicatorIcon-3"
            />
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              John Doe
            </p>
            <div />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-textColor-6"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 1H9v2h6V1zm-4 13h2V8h-2v6zm8.03-6.61l1.42-1.42c-.43-.51-.9-.99-1.41-1.41l-1.42 1.42C16.07 4.74 14.12 4 12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9 9-4.03 9-9c0-2.12-.74-4.07-1.97-5.61zM12 20c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1"
            />
          </div>
        </div>
      </div>
      <div
        class="makeStyles-btnGroup-4"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </span>
            Start recording
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="add new chart"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              fill="none"
              focusable="false"
              height="29"
              viewBox="0 0 29 29"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M29 4.5V7.5H24.5V12H21.5V7.5H17V4.5H21.5V0H24.5V4.5H29ZM24.5 25.5H3.5V4.5H12.5V1.5H3.5C1.85 1.5 0.5 2.85 0.5 4.5V25.5C0.5 27.15 1.85 28.5 3.5 28.5H24.5C26.15 28.5 27.5 27.15 27.5 25.5V16.5H24.5V25.5ZM18.5 16.5V22.5H21.5V16.5H18.5ZM12.5 22.5H15.5V10.5H12.5V22.5ZM9.5 22.5V13.5H6.5V22.5H9.5Z"
                fill="#535353"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="quit student session"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              fill="none"
              focusable="false"
              height="29"
              viewBox="0 0 29 29"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.9408 8.55L14.25 12.2408L10.5592 8.55L8.55 10.5592L12.2408 14.25L8.55 17.9408L10.5592 19.95L14.25 16.2592L17.9408 19.95L19.95 17.9408L16.2592 14.25L19.95 10.5592L17.9408 8.55ZM14.25 0C6.36975 0 0 6.36975 0 14.25C0 22.1302 6.36975 28.5 14.25 28.5C22.1302 28.5 28.5 22.1302 28.5 14.25C28.5 6.36975 22.1302 0 14.25 0ZM14.25 25.65C7.96575 25.65 2.85 20.5343 2.85 14.25C2.85 7.96575 7.96575 2.85 14.25 2.85C20.5343 2.85 25.65 7.96575 25.65 14.25C25.65 20.5343 20.5343 25.65 14.25 25.65Z"
                fill="#535353"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
